### ISSUE TITLE
KinD installer/Resource not present

### ISSUE THREAD

**Issue by ghost** [containers/podman-desktop#2794]
### Bug description

KinD installer/Resource not present therefore I'm not able to create a kubernetes cluster.

The "KinD Extension" is installed and status runningrnetes cluster with Kind in podman-desktop.

Podman-desktop: 1.0.1 and 1.1.1.0
Podman: 4.5.0 and 4.5.1

![image](https://github.com/containers/podman-desktop/assets/104419999/d8d9001b-8192-4a82-8a7a-64f34d57ac0d)

![image](https://github.com/containers/podman-desktop/assets/104419999/f2c4a0e3-28dd-4f52-b391-3aa5da000ae7)

![image](https://github.com/containers/podman-desktop/assets/104419999/e10d2842-3f82-4b4e-affd-d43c92f97dde)

### Operating system

Windows 10.0.19045 (22H2) (amd64)

### Version

1.1.0

**Comment by jeffmaury** [comment]
Out of curiosity, did you install the Kind binary from Podman Desktop ?

**Comment by ghost** [comment]
That's the problem, I can't install Kind from podman-desktop. Ther is no kind button on the podman-desktop status bar.
The Kind Extension comes with podman-desktop out-of-the-box, so there is nothing to install manually.

**Comment by ghost** [comment]
On Fedora 38 I don't have that problem.

**Comment by jeffmaury** [comment]
I was not able to reproduce on a brand new Win10 Pro testing machine. Can you have a look at the Podman Desktop log (Ctrl + Shift + I)

**Comment by ghost** [comment]
This is what I get with Ctrl + Shift + I, 
Not shure this is what you are after ...
![image](https://github.com/containers/podman-desktop/assets/104419999/656edae3-eef3-4f28-803f-7b737e3ff6bd)

**Comment by benoitf** [comment]
@hiahab if you can copy/paste all the content from 'console' that is in red (or all content, would be easier)

**Comment by ghost** [comment]
```
index.ts:1335 main ↪️ [lima] Could not find podman socket at C:\Users\my-user\.lima\podman\sock\podman.sock
(anonymous) @ index.ts:1335

index.ts:1335 main ↪️ Unable to fetch the available extensions: Timeout awaiting 'connect' for 500ms
(anonymous) @ index.ts:1335

index.ts:1335 main ↪️ [kind] HttpError: fetch failed
    at C:\Users\my-user\AppData\Local\Programs\podman-desktop\resources\app.asar\extensions\kind\builtin\kind.cdix\dist\extension.js:12:316002
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Fa (C:\Users\my-user\AppData\Local\Programs\podman-desktop\resources\app.asar\extensions\kind\builtin\kind.cdix\dist\extension.js:7:10552)
    at async Td (C:\Users\my-user\AppData\Local\Programs\podman-desktop\resources\app.asar\extensions\kind\builtin\kind.cdix\dist\extension.js:7692:4126)
    at async gz.activateExtension (C:\Users\my-user\AppData\Local\Programs\podman-desktop\resources\app.asar\packages\main\dist\index.cjs:91:35552)
    at async gz.loadExtension (C:\Users\my-user\AppData\Local\Programs\podman-desktop\resources\app.asar\packages\main\dist\index.cjs:91:29583)
    at async Promise.all (index 2)
    at async gz.start (C:\Users\my-user\AppData\Local\Programs\podman-desktop\resources\app.asar\packages\main\dist\index.cjs:91:27197)
    at async Dde.initExtensions (C:\Users\my-user\AppData\Local\Programs\podman-desktop\resources\app.asar\packages\main\dist\index.cjs:343:42532)
    at async C:\Users\my-user\AppData\Local\Programs\podman-desktop\resources\app.asar\packages\main\dist\index.cjs:370:2511
(anonymous) @ index.ts:1335
emit @ VM4 sandbox_bundle:2
onMessage @ VM4 sandbox_bundle:2

index.ts:1335 main ↪️ (node:21792) UnhandledPromiseRejectionWarning: Error: EINVAL: invalid argument, stat '//./pipe/'
(Use `Podman Desktop --trace-warnings ...` to show where the warning was created)
(anonymous) @ index.ts:1335
emit @ VM4 sandbox_bundle:2
onMessage @ VM4 sandbox_bundle:2

index.ts:1335 main ↪️ (node:21792) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 2)
(anonymous) @ index.ts:1335
emit @ VM4 sandbox_bundle:2
onMessage @ VM4 sandbox_bundle:2

index.ts:1335 main ↪️ [kind] HttpError: fetch failed
    at C:\Users\my-user\AppData\Local\Programs\podman-desktop\resources\app.asar\extensions\kind\builtin\kind.cdix\dist\extension.js:12:316002
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Tp.isAvailable (C:\Users\my-user\AppData\Local\Programs\podman-desktop\resources\app.asar\extensions\kind\builtin\kind.cdix\dist\extension.js:14:74997)
    at async Td (C:\Users\my-user\AppData\Local\Programs\podman-desktop\resources\app.asar\extensions\kind\builtin\kind.cdix\dist\extension.js:7692:4177)
    at async gz.activateExtension (C:\Users\my-user\AppData\Local\Programs\podman-desktop\resources\app.asar\packages\main\dist\index.cjs:91:35552)
    at async gz.loadExtension (C:\Users\my-user\AppData\Local\Programs\podman-desktop\resources\app.asar\packages\main\dist\index.cjs:91:29583)
    at async Promise.all (index 2)
    at async gz.start (C:\Users\my-user\AppData\Local\Programs\podman-desktop\resources\app.asar\packages\main\dist\index.cjs:91:27197)
    at async Dde.initExtensions (C:\Users\my-user\AppData\Local\Programs\podman-desktop\resources\app.asar\packages\main\dist\index.cjs:343:42532)
    at async C:\Users\my-user\AppData\Local\Programs\podman-desktop\resources\app.asar\packages\main\dist\index.cjs:370:2511
(anonymous) @ index.ts:1335
emit @ VM4 sandbox_bundle:2
onMessage @ VM4 sandbox_bundle:2

```

**Comment by ghost** [comment]
If i run 'Podman Desktop.exe' with the --trace-warnings flag I get

```
PS C:\Users\my-user\AppData\Local\Programs\podman-desktop> & '.\Podman Desktop.exe'  --trace-warnings
PluginSystem: received dom-ready event from the UI
Error while fetching API groups: Error: tunneling socket could not be established, statusCode=403
Error while fetching API groups: Error: tunneling socket could not be established, statusCode=403
Checking for update
Update for version 1.1.0 is not available (latest version: 1.1.0, downgrade is disallowed).
Delayed startup, flushing 8 events
System ready. Loading extensions...
Activating extension (podman-desktop.compose)
Activation extension (podman-desktop.compose) ended
Activating extension (podman-desktop.docker)
Activation extension (podman-desktop.docker) ended
Activating extension (podman-desktop.kind)
Activating extension (podman-desktop.kube-context)
[kube-context] starting extension kube-context
Activating extension (podman-desktop.lima)
[lima] Could not find podman socket at C:\Users\my-user\.lima\podman\sock\podman.sock
Activation extension (podman-desktop.lima) ended
Activating extension (podman-desktop.podman)
Activating extension (podman-desktop.registries)
Unable to fetch the available extensions: Timeout awaiting 'connect' for 500ms
Activation extension (podman-desktop.kube-context) ended
Activation extension (podman-desktop.registries) ended
Activation extension (podman-desktop.podman) ended
[21792:0613/093132.585:ERROR:CONSOLE(1)] "Uncaught (in promise) TypeError: Failed to fetch", source: devtools://devtools/bundled/panels/elements/elements.js (1)
[kind] Vt [HttpError]: fetch failed
    at C:\Users\my-user\AppData\Local\Programs\podman-desktop\resources\app.asar\extensions\kind\builtin\kind.cdix\dist\extension.js:12:316002
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Fa (C:\Users\my-user\AppData\Local\Programs\podman-desktop\resources\app.asar\extensions\kind\builtin\kind.cdix\dist\extension.js:7:10552)
    at async Td (C:\Users\my-user\AppData\Local\Programs\podman-desktop\resources\app.asar\extensions\kind\builtin\kind.cdix\dist\extension.js:7692:4126)
    at async gz.activateExtension (C:\Users\my-user\AppData\Local\Programs\podman-desktop\resources\app.asar\packages\main\dist\index.cjs:91:35552)
    at async gz.loadExtension (C:\Users\my-user\AppData\Local\Programs\podman-desktop\resources\app.asar\packages\main\dist\index.cjs:91:29583)
    at async Promise.all (index 2)
    at async gz.start (C:\Users\my-user\AppData\Local\Programs\podman-desktop\resources\app.asar\packages\main\dist\index.cjs:91:27197)
    at async Dde.initExtensions (C:\Users\my-user\AppData\Local\Programs\podman-desktop\resources\app.asar\packages\main\dist\index.cjs:343:42532)
    at async C:\Users\my-user\AppData\Local\Programs\podman-desktop\resources\app.asar\packages\main\dist\index.cjs:370:2511 {
  status: 500,
  request: {
    method: 'GET',
    url: 'https://api.github.com/repos/kubernetes-sigs/kind/releases',
    headers: {
      accept: 'application/vnd.github.v3+json',
      'user-agent': 'octokit-rest.js/19.0.11 octokit-core.js/4.2.1 Node.js/18.15.0 (win32; x64)'
    },
    request: { hook: [Function: bound bound Br] }
  }
}
(node:21792) UnhandledPromiseRejectionWarning: Error: EINVAL: invalid argument, stat '//./pipe/'
(Use `Podman Desktop --trace-warnings ...` to show where the warning was created)
(node:21792) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 2)
[kind] Vt [HttpError]: fetch failed
    at C:\Users\my-user\AppData\Local\Programs\podman-desktop\resources\app.asar\extensions\kind\builtin\kind.cdix\dist\extension.js:12:316002
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Tp.isAvailable (C:\Users\my-user\AppData\Local\Programs\podman-desktop\resources\app.asar\extensions\kind\builtin\kind.cdix\dist\extension.js:14:74997)
    at async Td (C:\Users\my-user\AppData\Local\Programs\podman-desktop\resources\app.asar\extensions\kind\builtin\kind.cdix\dist\extension.js:7692:4177)
    at async gz.activateExtension (C:\Users\my-user\AppData\Local\Programs\podman-desktop\resources\app.asar\packages\main\dist\index.cjs:91:35552)
    at async gz.loadExtension (C:\Users\my-user\AppData\Local\Programs\podman-desktop\resources\app.asar\packages\main\dist\index.cjs:91:29583)
    at async Promise.all (index 2)
    at async gz.start (C:\Users\my-user\AppData\Local\Programs\podman-desktop\resources\app.asar\packages\main\dist\index.cjs:91:27197)
    at async Dde.initExtensions (C:\Users\my-user\AppData\Local\Programs\podman-desktop\resources\app.asar\packages\main\dist\index.cjs:343:42532)
    at async C:\Users\my-user\AppData\Local\Programs\podman-desktop\resources\app.asar\packages\main\dist\index.cjs:370:2511 {
  status: 500,
  request: {
    method: 'GET',
    url: 'https://api.github.com/repos/kubernetes-sigs/kind/releases',
    headers: {
      accept: 'application/vnd.github.v3+json',
      'user-agent': 'octokit-rest.js/19.0.11 octokit-core.js/4.2.1 Node.js/18.15.0 (win32; x64)'
    },
    request: { hook: [Function: bound bound Br] }
  }
}
Activation extension (podman-desktop.kind) ended
PluginSystem: initialization done.
Autostarting container engine
[podman] Podman extension: Do not start a machine as there is already one starting or started
Installing Podman Desktop startup file at C:\Users\my-user\AppData\Roaming\Microsoft\Windows\Start Menu\Programs\Startup\podman-desktop.vbs using C:\Users\my-user\AppData\local\Programs\podman-desktop\Podman Desktop.exe location.


```

**Comment by benoitf** [comment]
@hiahab I guess you're behind a proxy as all http connections are not working

**Comment by ghost** [comment]
That's right.

**Comment by ghost** [comment]
So setting up proxy in Podman-desktop and in windows does not help...

**Comment by ghost** [comment]
Does podman-desktop reads windows environment?
- $env:HTTP_PROXY
- $env:HTTPS_PROXY
- $env:NO_PROXY

**Comment by benoitf** [comment]
@hiahab no it does not read env variables as mostly it's set in your .bashrc or .zshrc and `/Applications` on macOS are launched using default PATH/env (not reading anything from these env variables)

it could read system proxy

**Comment by themr0c** [comment]
I am not sure what from this thread could go in the troubleshooting guide. Is this wild guess appropriate?

On Windows, in an environment restricted by a proxy, Podman Desktop cannot install Kind (or Compose?).

The proxy settings in Podman Desktop only apply to the Podman machine.
Podman Desktop does not read environment variables such as `$env:HTTP_PROXY`, `$env:HTTPS_PROXY`, or `$env:NO_PROXY`.

Workaround

???

**Comment by benoitf** [comment]
> The proxy settings in Podman Desktop only apply to the Podman machine.

no it's not only for a podman machine, all extensions when performing http requests have proxy being set (1.3.1+)
https://github.com/containers/podman-desktop/pull/2825

**Comment by themr0c** [comment]
@hiahab Can you still reproduce the issue with Podman Desktop 1.3.1+?

**Comment by ghost** [comment]
I haven't tested Podman for a while on windows. Can do it during the next days.
But it works perfect in Linux :-)

**Comment by mrthomnas** [comment]
> @hiahab Can you still reproduce the issue with Podman Desktop 1.3.1+?

@themr0c As a user on Windows 11 behind a proxy, I can confirm this behavior on Podman Desktop 1.3.1, I used the airgap installer (podman-desktop-airgap-1.3.1-setup.exe) thinking that wouldn't result in this issue, but it does.

**Comment by themr0c** [comment]
I understand that we have to:

- Clarify what the airgap installer contains (now), and what you need to retrieve by other means, in https://podman-desktop.io/docs/Installation/windows-install/installing-podman-desktop-and-podman-in-a-restricted-environment.
- Clarify Kind and Compose installation steps, when the user is in a restricted environment.
- Change Podman Destkop behavior for binaries that Podman Desktop can't download, such as Kind, and Compose: add them in the airgap installer? Podman Desktop to use the proxy settings?

**Comment by ghost** [comment]
I havn't tested podman-desktop-airgap.
Hav been testing "normal" podman-desktop for windows (v10) but ... it's not going so good ....
Had to test some more.
Can also test the airgap installation.

**Comment by rommon** [comment]
I have the same issue.

I installed podman desktop from the airgap installation as I am in an environment with company proxy and I added podman machine with custom name and with network user mode enabled.

I also recognized that updating the "catalog extensions store" is also returning an error in the console log.

```
index.ts:1506 main ↪️ Unable to fetch the available extensions: getaddrinfo EAI_AGAIN registry.podman-desktop.io
console.<computed> @ index.ts:1506
(anonymous) @ index.ts:1506
emit @ VM4 sandbox_bundle:2
onMessage @ VM4 sandbox_bundle:2
```
![image](https://github.com/containers/podman-desktop/assets/3919401/bedf08fc-3690-4d3f-8a30-cbf0193298f6)

`kind` is not installed and there is also no button to click on.

clicking on the `compose` surface is also not doing anything

**Comment by rommon** [comment]
just tried the execution @hiahab was doing. this is my output:

```
PS C:\Users\rommon\AppData\Local\Programs\podman-desktop> & '.\Podman Desktop.exe'  --trace-warnings
PS C:\Users\rommon\AppData\Local\Programs\podman-desktop>
Kubeconfig path C:\Users\rommon\.kube\config provided does not exist. Skipping.
Checking for update
PluginSystem: received dom-ready event from the UI
Update for version 1.3.1 is not available (latest version: 1.3.1, downgrade is disallowed).
Delayed startup, flushing 7 events
System ready. Loading extensions...
Activating extension (podman-desktop.compose)
Activating extension (podman-desktop.docker)
Activating extension (podman-desktop.kind)
Activating extension (podman-desktop.kube-context)
[kube-context] starting extension kube-context
Activating extension (podman-desktop.lima)
[lima] Could not find socket at C:\Users\rommon\.lima\podman\sock\podman.sock
[lima] Could not find config at C:\Users\rommon\.lima\podman\copied-from-guest\kubeconfig.yaml
Activating extension (podman-desktop.podman)
Activating extension (podman-desktop.registries)
Activating extension (podman-desktop.compose) ended
Activating extension (podman-desktop.docker) ended
Activating extension (podman-desktop.kube-context) ended
Activating extension (podman-desktop.lima) ended
Unable to fetch the available extensions: getaddrinfo EAI_AGAIN registry.podman-desktop.io
Unable to fetch the available extensions: getaddrinfo EAI_AGAIN registry.podman-desktop.io
Activating extension (podman-desktop.registries) ended
[kind] ye [HttpError]: fetch failed
    at C:\Users\rommon\AppData\Local\Programs\podman-desktop\resources\app.asar\extensions\kind\builtin\kind.cdix\dist\extension.js:12:8268
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async hn (C:\Users\rommon\AppData\Local\Programs\podman-desktop\resources\app.asar\extensions\kind\builtin\kind.cdix\dist\extension.js:7:10552)
    at async fl (C:\Users\rommon\AppData\Local\Programs\podman-desktop\resources\app.asar\extensions\kind\builtin\kind.cdix\dist\extension.js:7678:629)
    at async $j.activateExtension (C:\Users\rommon\AppData\Local\Programs\podman-desktop\resources\app.asar\packages\main\dist\index.cjs:91:44909)
    at async $j.loadExtension (C:\Users\rommon\AppData\Local\Programs\podman-desktop\resources\app.asar\packages\main\dist\index.cjs:91:38182)
    at async Promise.all (index 2)
    at async $j.loadExtensions (C:\Users\rommon\AppData\Local\Programs\podman-desktop\resources\app.asar\packages\main\dist\index.cjs:91:34980)
    at async $j.start (C:\Users\rommon\AppData\Local\Programs\podman-desktop\resources\app.asar\packages\main\dist\index.cjs:91:34269)
    at async yhe.initExtensions (C:\Users\rommon\AppData\Local\Programs\podman-desktop\resources\app.asar\packages\main\dist\index.cjs:328:411493) {
  status: 500,
  request: {
    method: 'GET',
    url: 'https://api.github.com/repos/kubernetes-sigs/kind/releases',
    headers: {
      accept: 'application/vnd.github.v3+json',
      'user-agent': 'octokit-rest.js/20.0.1 octokit-core.js/5.0.0 Node.js/18.15.0 (win32; x64)'
    },
    request: { hook: [Function: bound bound gs] }
  }
}
[kind] ye [HttpError]: fetch failed
    at C:\Users\rommon\AppData\Local\Programs\podman-desktop\resources\app.asar\extensions\kind\builtin\kind.cdix\dist\extension.js:12:8268
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async fa.isAvailable (C:\Users\rommon\AppData\Local\Programs\podman-desktop\resources\app.asar\extensions\kind\builtin\kind.cdix\dist\extension.js:14:75453)
    at async fl (C:\Users\rommon\AppData\Local\Programs\podman-desktop\resources\app.asar\extensions\kind\builtin\kind.cdix\dist\extension.js:7678:680)
    at async $j.activateExtension (C:\Users\rommon\AppData\Local\Programs\podman-desktop\resources\app.asar\packages\main\dist\index.cjs:91:44909)
    at async $j.loadExtension (C:\Users\rommon\AppData\Local\Programs\podman-desktop\resources\app.asar\packages\main\dist\index.cjs:91:38182)
    at async Promise.all (index 2)
    at async $j.loadExtensions (C:\Users\rommon\AppData\Local\Programs\podman-desktop\resources\app.asar\packages\main\dist\index.cjs:91:34980)
    at async $j.start (C:\Users\rommon\AppData\Local\Programs\podman-desktop\resources\app.asar\packages\main\dist\index.cjs:91:34269)
    at async yhe.initExtensions (C:\Users\rommon\AppData\Local\Programs\podman-desktop\resources\app.asar\packages\main\dist\index.cjs:328:411493) {
  status: 500,
  request: {
    method: 'GET',
    url: 'https://api.github.com/repos/kubernetes-sigs/kind/releases',
    headers: {
      accept: 'application/vnd.github.v3+json',
      'user-agent': 'octokit-rest.js/20.0.1 octokit-core.js/5.0.0 Node.js/18.15.0 (win32; x64)'
    },
    request: { hook: [Function: bound bound gs] }
  }
}
Activating extension (podman-desktop.kind) ended
Activating extension (podman-desktop.podman) ended
PluginSystem: initialization done.
Autostarting container engine
[podman] Podman extension: Do not start a machine as there is already one starting or started
Unable to fetch the available extensions: getaddrinfo EAI_AGAIN registry.podman-desktop.io
Installing Podman Desktop startup file at C:\Users\rommon\AppData\Roaming\Microsoft\Windows\Start Menu\Programs\Startup\podman-desktop.vbs using C:\Users\rommon\AppData\local\Programs\podman-desktop\Podman Desktop.exe location.
Unable to fetch the available extensions: getaddrinfo EAI_AGAIN registry.podman-desktop.io
```

**Comment by jeffmaury** [comment]
Re opening as it's not only airgap related
